src/opm/parser/eclipse/EclipseState/Schedule/Group/Group.cpp

```diff
@@ -79,19 +78,21 @@ namespace Opm {
     }
 
     bool Group::isProductionGroup(size_t timeStep) const {
-        return bool( m_isProductionGroup.get(timeStep) );
+      return ((this->group_type.get(timeStep) & GroupType::PRODUCTION) == GroupType::PRODUCTION);
     }
 
     bool Group::isInjectionGroup(size_t timeStep) const {
-        return bool( m_isInjectionGroup.get(timeStep) );
+        return ((this->group_type.get(timeStep) & GroupType::INJECTION) == GroupType::INJECTION);
     }
 
-    void Group::setProductionGroup(size_t timeStep, bool isProductionGroup_) {
-        m_isProductionGroup.update(timeStep, isProductionGroup_);
+    void Group::setProductionGroup(size_t timeStep) {
+        auto gt = this->group_type.get(timeStep);
+        this->group_type.update(timeStep, gt | GroupType::PRODUCTION);
     }
```

Comment:
I think I'd prefer if this were written in terms of a helper function, e.g.,
```C++
this->setGroupTypeBit(timeStep, GroupType::PRODUCTION);
```
in which `setGroupTypeBit()` (or some other name) would be the obvious
```C++
void setGroupTypeBit(const std::size_t timeStep, const GroupType gtype)
{
    this->group_type.update(timeStep, this->group_type.get(timeStep) | gtype);
}
```
which you could reuse in `setInjectionGroup()` too. That would limit the amount of bit manipulation needed elsewhere.